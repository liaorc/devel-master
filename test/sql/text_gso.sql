--#
--#       Gpu Text Sort TestCases. 
--#  [TODO] Do not test commented-out queries until GPUSort supports TOAST data process. 
--#         If will support it, please remake expected outs and test_init.sql
--#

set pg_strom.debug_force_gpusort to on;
set pg_strom.gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set pg_strom.enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;


select * from (select row_number() over (order by char_x desc) as rowid,char_x from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by char_x asc) as rowid,char_x from strom_string_test) as t where t.rowid%100=0;

select * from (select row_number() over (order by nchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by nchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

select * from (select row_number() over (order by vchar_x desc) as rowid,vchar_x from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by vchar_x asc) as rowid,vchar_x from strom_string_test) as t where t.rowid%100=0;

-- select * from (select row_number() over (order by nvchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by nvchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

-- select * from (select row_number() over (order by text_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
