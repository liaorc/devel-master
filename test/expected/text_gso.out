--#
--#       Gpu Text Sort TestCases. 
--#  [TODO] Do not test commented-out queries until GPUSort supports TOAST data process. 
--#         If will support it, please remake expected outs and test_init.sql
--#
set pg_strom.debug_force_gpusort to on;
set pg_strom.gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set pg_strom.enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;
select * from (select row_number() over (order by char_x desc) as rowid,char_x from strom_string_test) as t where t.rowid%100=0;
 rowid | char_x 
-------+--------
   100 | e
   200 | c
   300 | b
   400 | 9
   500 | 8
   600 | 6
   700 | 4
   800 | 3
   900 | 1
  1000 | 0
(10 rows)

select * from (select row_number() over (order by char_x asc) as rowid,char_x from strom_string_test) as t where t.rowid%100=0;
 rowid | char_x 
-------+--------
   100 | 1
   200 | 3
   300 | 4
   400 | 6
   500 | 8
   600 | 9
   700 | b
   800 | c
   900 | e
  1000 | f
(10 rows)

select * from (select row_number() over (order by nchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 643
   200 | 233
   300 | 272
   400 | 516
   500 | 549
   600 | 181
   700 | 230
   800 | 475
   900 | 713
  1000 | 449
(10 rows)

select * from (select row_number() over (order by nchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 935
   200 | 585
   300 | 236
   400 | 778
   500 | 708
   600 |  37
   700 | 570
   800 | 104
   900 | 604
  1000 | 849
(10 rows)

select * from (select row_number() over (order by vchar_x desc) as rowid,vchar_x from strom_string_test) as t where t.rowid%100=0;
 rowid | vchar_x 
-------+---------
   100 | e
   200 | c
   300 | b
   400 | 9
   500 | 7
   600 | 6
   700 | 4
   800 | 3
   900 | 1
  1000 | 0
(10 rows)

select * from (select row_number() over (order by vchar_x asc) as rowid,vchar_x from strom_string_test) as t where t.rowid%100=0;
 rowid | vchar_x 
-------+---------
   100 | 1
   200 | 3
   300 | 4
   400 | 6
   500 | 7
   600 | 9
   700 | b
   800 | c
   900 | e
  1000 | f
(10 rows)

-- select * from (select row_number() over (order by nvchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by nvchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
